src/main/scala/edu/cmu/cs/obsidian/codegen/CodeGenYul.scala

```diff
@@ -214,12 +215,13 @@ object CodeGenYul extends CodeGenerator {
                             case NumLiteral(_) => LiteralKind.number
                             case TrueLiteral() => LiteralKind.boolean
                             case FalseLiteral() => LiteralKind.boolean
-                            case l =>
-                                assert(assertion = false, "TODO: unimplemented translate assignment case: " + l.toString)
+                            case StringLiteral(_) => LiteralKind.string
+                            case _ =>
+                                assert(assertion = false, s"unimplemented assignment case ${assignTo.toString}")
                                 LiteralKind.number
                         }
                         Seq(ExpressionStatement(FunctionCall(Identifier("sstore"),
-                            Seq(ilit(tempSymbolTable(x)), Literal(kind, e.toString, kind.toString)))))
+                            Seq(ilit(tempSymbolTable(x)), Literal(kind, e.toString, kind.toString))))) //todo: this is likely wrong for strings
```

Comment:
If you think it's wrong, assert rather than silently doing the wrong thing.

Reply:
i didn't want to make an assertion here because the code *runs* just fine with string constants, solc is happy to pretty print the ouput, but i think it won't actually work once we can check the output. so this is more of a note to myself for the future (e.g. #313) but i want to be able to check the output now.